### PR TITLE
virttest.qemu_storage: add support for rate limit

### PR DESCRIPTION
**Pull request by XueqiangWei**
1. add parameter -r to convert method
2. add parameter -r to commit method

Signed-off-by: Xueqiang Wei <xuwei@redhat.com>
ID: 1909738

### CODE REVIEW

**Comment by XueqiangWei** [comment]
(1/3) Host_RHEL.m8.u4.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_0.q35: PASS (576.02 s)
 (2/3) Host_RHEL.m8.u4.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_4611686018427387400.q35: PASS (546.72 s)
 (3/3) Host_RHEL.m8.u4.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_9223372036854775295.q35: PASS (503.36 s)

(1/4) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.commit_with_backing: PASS (195.11 s)
 (2/4) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.commit_with_backing.rate_limit.with_rate_limit_0: PASS (184.37 s)
 (3/4) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.commit_with_backing.rate_limit.with_rate_limit_4611686018427387400: PASS (174.60 s)
 (4/4) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.commit_with_backing.rate_limit.with_rate_limit_9223372036854775295: PASS (171.98 s)

**Comment by XueqiangWei** [comment]
@zhencliu Please help review it. Many thanks.

File: virttest/qemu_storage.py [hunk 1]
```diff
@@ -414,6 +414,8 @@ class QemuImg(storage.QemuImg):
         "source_cache_mode": "-T",
         "target_image_format": "-O",
         "convert_sparse_size": "-S",
+        "convert_rate_limit": "-r",
+        "commit_rate_limit": "-r",
```
Comment (zhencliu) [inline virttest/qemu_storage.py:418]: what about add only one item, e.g. rate_limit: "-r", for both convert_rate_limit and commit_rate_limit is the same option

File: virttest/qemu_storage.py [hunk 2]
```diff
@@ -822,17 +826,22 @@ def convert(self, params, root_dir, cache_mode=None,
...
+                indicate rate limit for the convert process,
+                the unit is bytes per second
         """
         convert_target = params["convert_target"]
         convert_params = params.object_params(convert_target)
         convert_image = QemuImg(convert_params, root_dir, convert_target)
 
         convert_compressed = convert_params.get("convert_compressed")
         sparse_size = convert_params.get("sparse_size")
+        rate_limit = convert_params.get("rate_limit")
```
Comment (zhencliu) [inline virttest/qemu_storage.py:839]: add a prefix image, like image_rate_limit?

File: virttest/qemu_storage.py (hunk 1 above)
Comment (XueqiangWei) [reply virttest/qemu_storage.py:418]: Agree, I will update them.

**Comment by XueqiangWei** [comment]
(1/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_raw.rate_limit.with_rate_limit_0: PASS (215.30 s)
 (2/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_raw.rate_limit.with_rate_limit_4611686018427387400: PASS (199.45 s)
 (3/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_raw.rate_limit.with_rate_limit_9223372036854775295: PASS (209.84 s)
 (4/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_0: PASS (230.66 s)
 (5/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_4611686018427387400: PASS (227.06 s)
 (6/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_9223372036854775295: PASS (229.73 s)
 (7/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_luks.rate_limit.with_rate_limit_0: PASS (279.99 s)
 (8/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_luks.rate_limit.with_rate_limit_4611686018427387400: PASS (283.86 s)
 (9/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_luks.rate_limit.with_rate_limit_9223372036854775295: PASS (286.44 s)

(1/4) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.commit_with_backing: PASS (171.40 s)
 (2/4) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.commit_with_backing.rate_limit.with_rate_limit_0: PASS (160.09 s)
 (3/4) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.commit_with_backing.rate_limit.with_rate_limit_4611686018427387400: PASS (156.03 s)
 (4/4) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.commit_with_backing.rate_limit.with_rate_limit_9223372036854775295: PASS (146.97 s)

@zhencliu updated, please help review it again. Many thanks.

**Comment by zhencliu** [comment]
LGTM

**Comment by XueqiangWei** [comment]
@YongxueHong please help review it, many thanks.

**Review by YongxueHong — APPROVED** [review]
LGTM.

**Pull request merged**